### PR TITLE
Added tutorial for dev environment setup

### PR DESCRIPTION
**Pull request by fvigo**
## Status
Ready

## Description
Added a tutorial for setting up a development environment.

### CODE REVIEW

File: docs/tutorial-setup-dev.md [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+So far we've been using the following Operating Systems:
+- MacOS
+- Linux
+- Windows (only with WSL - [Windows Subsystem For Linux](https://docs.microsoft.com/en-us/windows/wsl/about))
+
+If you successfully manage to get this work on other platforms (native Windows, OpenBSD, etc.) , please let us know and we'll add it to the tutorial! (click on Report an issue at the bottom of this page).
+
+### Python
+
+You will need to build your integration using **Python** and, more specifically, Python 3.6+. While some content is built via Javascript and Python 2, we require Python 3.6+ for contributions.
```
Comment (glicht) [inline docs/tutorial-setup-dev.md]:
```suggestion
You will need to build your integration using **Python** and, more specifically, Python 3.7+. While some content is built via Javascript and Python 2, we require Python 3.7+ for contributions.
```

Let's go with python 3.7. I think we started using some features of 3.7 in some our scripts.

File: docs/tutorial-setup-dev.md [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+
+You will need a **[GitHub](https://github.com)** account, as the contribution process requires you to submit a Pull Request in the [Demisto Content Repository](https://github.com/demisto/content). To learn more about Pull Requests and contributing , check out the [Collaborating with issue and pull requests](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests) tutorial on GitHub, as well as our [Content Contribution Guide](https://github.com/demisto/content/blob/master/CONTRIBUTING.md).
+
+And you will need a `git` client on your system (git, GitHub Desktop, SourceTree, etc). In the examples we'll just use the `git` command line client.
+
+### Docker
+
+In order to be able to run linting and tests, you should have **Docker** installed on your machine. This way you can test your code using the same Python environment as the one that will run inside the Demisto instance.
+
+*Note* if you're using WSL 1, you cannot run Docker natively on WSL, but you can install Docker Desktop on Windows and configure WSL to communicate to it using [this](https://nickjanetakis.com/blog/setting-up-docker-for-windows-and-wsl-to-work-flawlessly) tutorial.
```
Comment (glicht) [inline docs/tutorial-setup-dev.md]:
```suggestion
*Note* if you're using WSL, you cannot run Docker natively on WSL, but you can install Docker Desktop on Windows and configure WSL to communicate to it using [this](https://nickjanetakis.com/blog/setting-up-docker-for-windows-and-wsl-to-work-flawlessly) tutorial.
```

File: docs/tutorial-setup-dev.md [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+Not much to check here, just go to [GitHub](https://github.com) and make sure that you have an account or Sign Up for one:
+
+![GitHub](doc_imgs/03-github.png)
+
+#### Docker
+
+Make sure that `docker` is installed on your system and is working correctly by running the `hello-world` container:
+
+```bash
+sb@dddd:~/demisto$ docker run hello-world
```
Comment (glicht) [inline docs/tutorial-setup-dev.md]:
```suggestion
sb@dddd:~/demisto$ docker run --rm hello-world
```

Add --rm so the container is deleted when done.

File: docs/tutorial-setup-dev.md [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+ 1. The Docker client contacted the Docker daemon.
+ 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
+    (amd64)
+ 3. The Docker daemon created a new container from that image which runs the
+    executable that produces the output you are currently reading.
+ 4. The Docker daemon streamed that output to the Docker client, which sent it
+    to your terminal.
+
+To try something more ambitious, you can run an Ubuntu container with:
+ $ docker run -it ubuntu bash
```
Comment (glicht) [inline docs/tutorial-setup-dev.md]:
```suggestion
 $ docker run --rm -it ubuntu bash
```

File: docs/tutorial-setup-dev.md [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+Share images, automate workflows, and more with a free Docker ID:
+ https://hub.docker.com/
+
+For more examples and ideas, visit:
+ https://docs.docker.com/get-started/
+
+sb@dddd:~/demisto$
+```
+
+*Note:* if you're using Windows with WSL 1, you can still use Docker Desktop from WSL. Follow [this](https://nickjanetakis.com/blog/setting-up-docker-for-windows-and-wsl-to-work-flawlessly) tutorial for details.
```
Comment (glicht) [inline docs/tutorial-setup-dev.md]:
```suggestion
*Note:* if you're using Windows with WSL, you can still use Docker Desktop from WSL. Follow [this](https://nickjanetakis.com/blog/setting-up-docker-for-windows-and-wsl-to-work-flawlessly) tutorial for details.
```

File: docs/tutorial-setup-dev.md [hunk 6]
```diff
@@ -0,0 +1,464 @@
...
+remote: Counting objects: 100% (108/108), done.
+remote: Compressing objects: 100% (90/90), done.
+remote: Total 101143 (delta 50), reused 53 (delta 18), pack-reused 101035
+Receiving objects: 100% (101143/101143), 110.65 MiB | 11.04 MiB/s, done.
+Resolving deltas: 100% (73634/73634), done.
+Checking out files: 100% (4522/4522), done.
+sb@dddd:~/demisto$
+```
+
+*Note:* you must clone **your fork** of the repository, as you will need to be able to write into it. Do **not** clone `demisto/content`, as you won't be able to commit.
```
Comment (glicht) [inline docs/tutorial-setup-dev.md]:
```suggestion
*Note:* you must clone **your fork** of the repository, as you will need to be able to write into it. Do **not** clone `demisto/content`, as you won't be able to push commits.
```

**Comment by anara123** [comment]
@glicht @ronykoz after resent changes to demisto-sdk, do we still need python2 installed?

**Comment by anara123** [comment]
@fvigo overall looks good!
If you show `demisto-sdk lint`, maybe it is a good idea to show `demisto-sdk validate` too.
But some of the documentation we will have to change a bit very soon. We already support Packs.

**Comment by ronykoz** [comment]
@anara123 i don't think that we really need the python2, maybe just for working on old integrations that use that, so i would keep it in the meantime

**Comment by fvigo** [comment]
> @fvigo overall looks good!
> If you show `demisto-sdk lint`, maybe it is a good idea to show `demisto-sdk validate` too.
> But some of the documentation we will have to change a bit very soon. We already support Packs.

Thanks. I added a reference to `demisto-sdk validate` too. We'll update this tutorial as we go.

**Comment by glicht** [comment]
Looks good

**Pull request merged**